Havtorn/Source/Engine/Core/Color.h

```diff
@@ -0,0 +1,106 @@
+// Copyright 2022 Team Havtorn. All Rights Reserved.
+
+#pragma once
+
+namespace Havtorn
+{
+	// Represents a color in RGBA 0-255 range.
+	struct HAVTORN_API SColor
+	{
+		U8 R, G, B, A = 255;
+
+		static const SColor Red;
+		static const SColor Green;
+		static const SColor Blue;
+		static const SColor Black;
+		static const SColor White;
+		static const SColor Grey;
+		static const SColor Teal;
+		static const SColor Orange;
+		static const SColor Magenta;
+		static const SColor Yellow;
+		// TODO.AG: Add more color presets
+
+		inline SColor();
+		// Alpha is set to 255.
+		inline SColor(U8 monochrome);
+		inline SColor(U8 r, U8 g, U8 b);
+		inline SColor(U8 r, U8 g, U8 b, U8 a);
+		// Expects SVector values to be [0.0f - 1.0f] range.
+		inline SColor(const SVector& rgb);
+		// Expects SVector4 values to be [0.0f - 1.0f] range.
+		inline SColor(const SVector4& rgba);
+
+		// Returns the RGB values in a [0.0f - 1.0f] range.
+		inline SVector AsVector() const;
+		// Returns the RGBA values in a [0.0f - 1.0f] range.
+		inline SVector4 AsVector4() const;
+	};
+
+	SColor::SColor()
```

Comment:
I think it's nicer here to put all these values in the initializer lists, the nice thing is that you don't have to assign any of the values that are using the default values since they are initialized within the class